chat/views.py

```diff
@@ -27,19 +28,27 @@ def get_context_data(self, **kwargs):
             vector = SearchVector('name', 'context')
 
             chats = Chat.objects.annotate(
-                rank=SearchRank(vector, search_query),
-                headline_name=SearchHeadline('name', search_query, start_sel='<mark>', stop_sel='</mark>'),
-                headline_context=SearchHeadline('context', search_query, start_sel='<mark>', stop_sel='</mark>')
+                rank=SearchRank(vector, search_query)
             ).filter(
                 Q(rank__gte=0.001) | Q(message__parts__icontains=query)
             ).order_by('-rank').distinct()
 
             messages = Message.objects.filter(
                 Q(parts__icontains=query),
                 Q(chat__user=self.request.user)
-            ).select_related('chat').annotate(
-                headline_parts=SearchHeadline('parts', search_query, start_sel='<mark>', stop_sel='</mark>')
-            ).distinct()
+            ).select_related('chat').distinct()
+
+            def highlight(text, search_query):
+                if not text:
+                    return ''
+                pattern = re.compile(re.escape(search_query), re.IGNORECASE)
+                return pattern.sub(f'<mark>{search_query}</mark>', text)
```

Comment:
can move this function definition out of the function.

Also not gonna highlight if its not an exact match but thats chilled for now.

Reply:
Moved the function definition to the  outside

- Will resolve the Full text search for Cockroach Db